include/os/freebsd/spl/sys/taskq.h

```diff
@@ -93,13 +94,16 @@ extern void taskq_dispatch_ent(taskq_t *, task_func_t, void *, uint_t,
     taskq_ent_t *);
 extern int taskq_empty_ent(taskq_ent_t *);
 taskq_t	*taskq_create(const char *, int, pri_t, int, int, uint_t);
+taskq_t	*taskq_create_synced(const char *, int, pri_t, int, int, uint_t,
+    kthread_t ***);
 taskq_t	*taskq_create_instance(const char *, int, int, pri_t, int, int, uint_t);
 taskq_t	*taskq_create_proc(const char *, int, pri_t, int, int,
     struct proc *, uint_t);
 taskq_t	*taskq_create_sysdc(const char *, int, int, int,
     struct proc *, uint_t, uint_t);
 void	nulltask(void *);
 extern void taskq_destroy(taskq_t *);
+extern void taskq_destroy_synced(taskq_t *, kthread_t **);
```

Comment:
I'm not 100% convinced we need a `taskq_destroy_synced()` function.  Why not allow the kthread_t array to be freed at any time by the caller with a `kmem_free()` call.  This seems like it would more naturally fit with how we want to use it, and would entirely remove the need for `spa->spa_kthreads`.   A local variable could be used.  Thoughts?

Reply:
updated, thanks!